connectors/src/lib/temporal_monitoring.ts

```diff
@@ -73,6 +77,22 @@ export class ActivityInboundLogInterceptor
 
       // eslint-disable-next-line @typescript-eslint/no-explicit-any
     } catch (err: any) {
+      if (err instanceof ExternalOauthTokenError) {
+        // We have a connector working on an expired token, we need to cancel the workflow.
+        const workflowId = this.context.info.workflowExecution.workflowId;
+        const client = await getTemporalClient();
+        try {
```

Comment:
Ideally we store that information on the connector so that it can be presented client-side? A simple status will do?